lightning/src/ln/channelmanager.rs

```diff
@@ -7220,38 +7187,53 @@ where
 		let mut peer_state = peer_state_opt.expect("peer_state_opt is always Some when the counterparty_node_id is Some");
 
 		let mut preimage_update = ChannelMonitorUpdate {
-			update_id: 0, // set in set_closed_chan_next_monitor_update_id
-			counterparty_node_id: prev_hop.counterparty_node_id,
+			update_id: 0, // set below
```

Comment:
Could you create `preimage_update` after calculating the `update_id` to avoid this?